### PR TITLE
Add How To guide for deployment plans

### PR DESCRIPTION
**Pull request by mefrem**
## Description

Deployment plans are a critical component of Clarinet, but documentation has been lacking. This document describes how deployment plans work, how to specify them, and how to manually edit and extend them.

Relevant issues: 
- https://github.com/hirosystems/docs/issues/247
- https://github.com/hirosystems/clarinet/issues/747

## Tasks remaining

- [ ] Deleting the Feature Guide for deployment plans
- [ ] Include a code example of `emulated-contract-call` deployment plan transaction
- [ ] A code example of "sending STX" via a deployment plan (confirm if it is a transaction or .toml file specification)
- [ ] A code example of "waiting for block" (ditto)
- [ ] Confirm no missing major features and limitations of different Stacks networks in table

## Checklist

- [ ] Link all relevant issues
- [ ] [Conventional commits were used](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] New links to files and images were verified
- [ ] For fixes/refactors: all existing references were identified and replaced
- [ ] [Style guide](https://developers.google.com/style) was reviewed and applied
- [x] Clear code samples were provided
- [x] People were tagged for review

Fixes https://github.com/hirosystems/docs/issues/247

### CODE REVIEW

**Review by lgalabru — APPROVED** [review]
Excellent, thank you so much for this documentation @mefrem 🙏

**Comment by mefrem** [comment]
Deployment plans documentation is complete and ready for review @LakshmiLavanyaKasturi

File: docs/clarinet/how-to-guides/how-to-use-deployment-plans.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+## Editing your deployment plan directly
+
+Recall that a default deployment plan, comprising your projects specifications as set in your `clarinet.toml` and `devnet.toml` files, is used for every deployment and is generated upon each `clarinet deployments generate --<network>` shell command to your `/deployments` folder.
+
+However, you can also manually edit this default deployment `.yaml` file directly—programmatically orchestrating your contract publication and interaction beyond your plan's defaults—with these `transaction` types and their fields available: 
+
+:::note
+If your deployment plan differs from the default plan that is a e
+If you manually configure the generated deployment plan under `/deployments`, Clarinet prompts you to overwrite it with the default deployment plan generated by your Clarinet project's `.toml` files—type `no` to proceed with your manually configured deployment plan.
```
Comment (LakshmiLavanyaKasturi) [inline docs/clarinet/how-to-guides/how-to-use-deployment-plans.md:71]: Split this sentence, maybe? 

Suppose you manually configure the generated deployment plan under `/deployments`. In that case, Clarinet prompts you to overwrite it with the default deployment plan generated by your Clarinet project's `.toml` files—type `no` to proceed with your manually configured deployment plan.

File: docs/clarinet/how-to-guides/how-to-use-deployment-plans.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        remap-sender: ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM
+        remap-principals:
+            SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE: ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM
+        cost: 8400
+        path: ".cache/requirements/SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.clar"
+```
+
+### `contract-publish`
+
+This is a default transaction that is included in every generated deployment plan for every contract your Clarinet project has. This "counter" application has a [devnet deployment plan](https://github.com/hirosystems/clarinet/blob/develop/components/clarinet-cli/examples/counter/deployments/default.devnet-plan.yaml) that publishes its core smart contract.
```
Comment (LakshmiLavanyaKasturi) [inline docs/clarinet/how-to-guides/how-to-use-deployment-plans.md:128]: This default transaction is included in every..

**Comment by LakshmiLavanyaKasturi** [comment]
> Deployment plans documentation is complete and ready for review @LakshmiLavanyaKasturi

Thanks for putting this together. Everything looks good. I just left a few minor comments. 

Also, After we merged remote content re-use PR, all the docs reside in the product-specific repos. It means that this file should be going into Clarinet repo   Clarinet/docs/how-to-guides/deployment-plans.md.  (https://github.com/hirosystems/clarinet/tree/develop/docs/how-to-guides).

Please let me know if you need help pushing this PR to the clarinet repo. Happy to help. Thanks again!

**Comment by mefrem** [comment]
Accepted your suggestions Lavanya, and submitted the file here for record's sake, and also to the Clarinet product repo.

https://github.com/hirosystems/clarinet/pull/798

**Comment by LakshmiLavanyaKasturi** [comment]
Closing this as a duplicate of https://github.com/hirosystems/docs/pull/298. Now that we have the docusaurus plugin that pulls remote files directly to Docs, we don't need this PR anymore.

**Pull request closed without merging**